src/Axes/DiscreteAxis.jl

```diff
@@ -309,6 +310,30 @@ function merge_close_ticks(v::AbstractVector{T}; min_diff::T = T(1e-6)) where {T
     v[1:n]
 end
 
+"""
+    merge_second_order_important_points(imp::Vector{T}, imp_second_order::Vector{T}; min_diff::T = T(1e-6)) where {T}
+
+Merges and sorts two vectors into one sorted vector. But only add elements of the second vector, `imp_second_order`, if
+they are not too close (via `min_diff`) to elements of the first vector. 
+"""
+function merge_second_order_important_points(imp::Vector{T}, imp_second_order::Vector{T}; min_diff::T = T(1e-6)) where {T}
+    sorted_imp = sort(imp)
+    all_merged = unique!(sort!(vcat(sorted_imp, imp_second_order)))
+    Δall_merged = diff(all_merged)
+    inds_diff_too_small = findall(Δ -> Δ < min_diff, Δall_merged)
+    delete_inds = Int[]
+    for i in inds_diff_too_small
+        if insorted(all_merged[i], sorted_imp)
+            push!(delete_inds, i+1)
+        elseif insorted(all_merged[i+1], sorted_imp)
+            push!(delete_inds, i)
+        else
+            error("This should never happen.") # I will leave this for now just to be sure.
+        end
+    end

```

Comment:
Is the method using `push!` the most efficient?

If we stick to your original docstring, and we only add the ticks of the second array if they fit (rather then adding all of them and removing them afterwars), can we do something like (untested):
```julia
sorted_imp_1 = sort(imp)
sorted_imp_2 = sort(imp_second_order)
final = Vector{T}(undef, length(sorted_imp1) + length(sorted_imp_2))
i1 = 1 # index for sorted_imp_1
i2 = 1 # index for sorted_imp_2
for i in eachindex(final)
    s1 = sorted_imp_1[i1]
    s2 = sorted_imp_2[i2]
    if s1 < s2
        final[i] = s1
        i1 += 1
        if s2 - s1 < min_diff
            i2 += 1
        end
    else
        if s1 - s2 < min_diff
            final[i] = s2
        else
            final[i] = s1
            i1 += 1
        end
        i2 += 1
    end
end
```
?